libdrgn/python/module.c

```diff
@@ -10,6 +10,52 @@ PyObject *FaultError;
 PyObject *MissingDebugInfoError;
 PyObject *OutOfBoundsError;
 
+int FaultError_init(PyObject *self, PyObject *args, PyObject *kwds)
+{
+	PyObject *address, *message;
+
+	if (!_PyArg_NoKeywords(Py_TYPE(self)->tp_name, kwds))
+		return -1;
+
+	if (!PyArg_ParseTuple(args, "OO:FaultError", &message, &address))
+		return -1;
+
+	if (PyObject_SetAttrString(self, "message", message) < 0)
+		return -1;
+
+	if (PyObject_SetAttrString(self, "address", address) < 0)
+		return -1;
+
+	return 0;
+}
+
+static PyObject *FaultError_str(PyObject *self)
+{
+	PyObject *message, *address;
+
+	message = PyObject_GetAttrString(self, "message");
+	if (!message)
+		return NULL;
+
+	address = PyObject_GetAttrString(self, "address");
+	if (!address)
+		return NULL;
+
+	return PyUnicode_FromFormat("%S: %S",
+				    message, PyNumber_ToBase(address, 16));
+}
+
+static PyTypeObject PyExc_FaultError = {
```

Comment:
I should've mentioned this on the last revision, the `PyExc_` prefix makes it look like a standard Python error. We can call this `FaultError_type`. Also, the definition of `FaultError_type` and its methods would make more sense in `error.c`.